File: src/data/convert_kospi_data.js
const fs = require('fs');
const path = require('path');

// KOSPI 회사 데이터 로드
const kospiCompaniesPath = path.join(__dirname, 'KOSPI_companies.json');
const kospiCompanies = JSON.parse(fs.readFileSync(kospiCompaniesPath, 'utf8'));

console.log('📊 KOSPI 데이터 분석:');
console.log(`총 기업 수: ${kospiCompanies.length}`);

// 산업별 기업 수 계산
const industryCount = {};
kospiCompanies.forEach(company => {
  const industry = company.industry;
  industryCount[industry] = (industryCount[industry] || 0) + 1;
});

console.log('\n📈 산업별 기업 수:');
Object.entries(industryCount)
  .sort((a, b) => b[1] - a[1])
  .forEach(([industry, count]) => {
    console.log(`  ${industry}: ${count}개`);
  });

// main_product가 없는 기업들 확인
const noMainProduct = kospiCompanies.filter(company => 
  !company.main_product || 
  company.main_product === null || 
  company.main_product === 'NaN' ||
  company.main_product === ''
);

console.log(`\n⚠️ main_product가 없는 기업: ${noMainProduct.length}개`);
if (noMainProduct.length > 0) {
  console.log('예시:');
  noMainProduct.slice(0, 5).forEach(company => {
    console.log(`  ${company.name} (${company.ticker}): ${company.main_product}`);
  });
}

// kospi_enriched_final.ts 파일 생성
const enrichedData = {};
kospiCompanies.forEach(company => {
  // ticker에서 .KS 제거하여 키로 사용
  const ticker = company.ticker.replace('.KS', '');
  
  enrichedData[ticker] = {
    name: company.name,
    description: company.main_product && 
                 company.main_product !== null && 
                 company.main_product !== 'NaN' && 
                 company.main_product !== '' 
                 ? company.main_product 
                 : `${company.industry} 관련 사업`,
    industry: company.industry
  };
});

// TypeScript 파일로 출력
const tsContent = `export const KOSPI_ENRICHED_FINAL = ${JSON.stringify(enrichedData, null, 2)} as const;`;

const outputPath = path.join(__dirname, 'kospi_enriched_final.ts');
fs.writeFileSync(outputPath, tsContent, 'utf8');

console.log(`\n✅ kospi_enriched_final.ts 파일 생성 완료`);
console.log(`📁 위치: ${outputPath}`);
console.log(`📊 총 ${Object.keys(enrichedData).length}개 기업 데이터 변환 완료`);

File: src/data/convert_kospi_data.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import json
import os
from collections import defaultdict

def convert_kospi_data():
    """KOSPI 데이터를 enriched_final 형태로 변환"""
    
    # 현재 스크립트 위치
    script_dir = os.path.dirname(os.path.abspath(__file__))
    
    # KOSPI 회사 데이터 로드
    kospi_companies_path = os.path.join(script_dir, 'KOSPI_companies.json')
    
    print("📊 KOSPI 데이터 변환 시작...")
    
    with open(kospi_companies_path, 'r', encoding='utf-8') as f:
        kospi_companies = json.load(f)
    
    print(f"총 기업 수: {len(kospi_companies)}")
    
    # 산업별 기업 수 계산
    industry_count = defaultdict(int)
    for company in kospi_companies:
        industry_count[company['industry']] += 1
    
    print("\n📈 산업별 기업 수 (상위 10개):")
    sorted_industries = sorted(industry_count.items(), key=lambda x: x[1], reverse=True)
    for industry, count in sorted_industries[:10]:
        print(f"  {industry}: {count}개")
    
    # main_product가 없는 기업들 확인
    no_main_product = []
    for company in kospi_companies:
        main_product = company.get('main_product')
        if not main_product or main_product in [None, 'NaN', '', 'null']:
            no_main_product.append(company)
    
    print(f"\n⚠️ main_product가 없는 기업: {len(no_main_product)}개")
    if no_main_product:
        print("예시:")
        for company in no_main_product[:5]:
            print(f"  {company['name']} ({company['ticker']}): {company.get('main_product', 'None')}")
    
    # kospi_enriched_final.ts 파일 생성
    enriched_data = {}
    for company in kospi_companies:
        # ticker에서 .KS 제거하여 키로 사용
        ticker = company['ticker'].replace('.KS', '')
        
        # main_product 처리
        main_product = company.get('main_product')
        if main_product and main_product not in [None, 'NaN', '', 'null']:
            description = str(main_product)
        else:
            description = f"{company['industry']} 관련 사업"
        
        enriched_data[ticker] = {
            "name": company['name'],
            "description": description,
            "industry": company['industry']
        }
    
    # TypeScript 파일로 출력
    ts_content = f"export const KOSPI_ENRICHED_FINAL = {json.dumps(enriched_data, ensure_ascii=False, indent=2)} as const;"
    
    output_path = os.path.join(script_dir, 'kospi_enriched_final.ts')
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(ts_content)
    
    print(f"\n✅ kospi_enriched_final.ts 파일 생성 완료")
    print(f"📁 위치: {output_path}")
    print(f"📊 총 {len(enriched_data)}개 기업 데이터 변환 완료")
    
    return len(enriched_data), len(sorted_industries)

if __name__ == "__main__":
    convert_kospi_data()
